packages/@react-spectrum/s2/src/Picker.tsx

```diff
@@ -394,26 +394,35 @@ export const Picker = /*#__PURE__*/ (forwardRef as forwardRefType)(function Pick
                 estimatedHeadingHeight: 50,
                 padding: 8,
                 loaderHeight: LOADER_ROW_HEIGHTS[size][scale]}}>
-              <PopoverBase
+              <Popover
                 hideArrow
                 offset={menuOffset}
+                crossOffset={isQuiet ? -12 : undefined}
                 placement={`${direction} ${align}` as Placement}
                 shouldFlip={shouldFlip}
                 UNSAFE_style={{
-                  width: menuWidth && !isQuiet ? `${menuWidth}px` : undefined
+                  width: menuWidth && !isQuiet ? `calc(${menuWidth}px - 2 * var(--s2-container-border-width))` : undefined
                 }}
+                // TODO: not sure how best to type styles so it also can accept arbitrary css vars
+                // @ts-ignore
                 styles={style({
-                  marginStart: {
-                    isQuiet: -12
+                  '--cross-offset': {
+                    type: 'width',
+                    value: -12
```

Comment:
Open to any suggestions on how best to type styles in Popover to accept arbitrary values for setting css variables, might be forgetting where we do this. From what I can tell, this pattern usually happens when setting `className` which accepts a style string

Reply:
is this?
> // TODO: not sure how best to type styles so it also can accept arbitrary css vars

you'd need to set it in UNSAFE_style, otherwise you need to specify every possible value and take it as an argument

Reply:
yeah, but setting it in `UNSAFE_style` doesn't allow for the above syntax and thus we'd need to manually transform that value to rems (or for cases like `borderOffsetWidth`, the value is actually in px) essentially duplicating a bunch of the work done in the theme